### PR TITLE
Excludes hardhat-diamond-abi/.* so no reprocessing

### PR DESCRIPTION
**Pull request by SuperGNUS**
This can be accomplished with the hardhat-config.ts file as well, but this is more transparent.

    exclude: ["hardhat-diamond-abi\/.*"],

### CODE REVIEW

**Comment by SuperGNUS** [comment]
Fixes #7

File: src/index.ts [hunk 1]
```diff
@@ -194,6 +194,11 @@ export async function generateDiamondAbi(
       log(`Skipping ${contractName} because it did matched an \`exclude\` pattern.`);
       continue;
     }
+    
+    // this should be the output filename, but this will work too
+    if (contractName.match("hardhat-diamond-abi\/.*")) {
```
Comment (phated) [inline src/index.ts]:
```suggestion
    if (contractName.startsWith(PLUGIN_NAME)) {
```
Wondering if it should be like this, which is more resilient to changing the plugin name and doesn't do another match (I believe startsWith is faster.

I also think I want to write a test for this, and log that it was skipped.

**Comment by phated** [comment]
I'm planning to add this test today and then I'll get this merged. Sorry for the delay!

**Comment by phated** [comment]
Thanks for sending this @GVTopCoder! I landed your commit with #10 but I needed to create a new branch because I couldn't push to yours to add tests and logging.

**Pull request closed without merging**